dapp/src/components/AccountListener.js

```diff
@@ -74,7 +76,26 @@ const AccountListener = (props) => {
     },
   })
 
-  const historyQuery = useTransactionHistoryQuery(account)
+  const historyPageQuery = useTransactionHistoryPageQuery(
+    account,
+    transactionHistoryItemsPerPage
+  )
+
+  const transactionItems = useMemo(
+    () =>
+      historyPageQuery.isSuccess
+        ? historyPageQuery.data.page.pages * transactionHistoryItemsPerPage
```

Comment:
Last page might not be complete (have the whole `transactionHistoryItemsPerPage`). Not sure if this plays a role